front/src/ctfnote/parsers/htb.ts

```diff
@@ -0,0 +1,79 @@
+import { ParsedTask, Parser } from '.';
+import { parseJson, parseJsonStrict } from '../utils';
+
+// output of https://ctf.hackthebox.com/api/public/challengeCategories
+const challengeCategories: { [index: number]: string } = {
+  1: 'Fullpwn',
+  2: 'Web',
+  3: 'Pwn',
+  4: 'Crypto',
+  5: 'Reversing',
+  6: 'Stego',
+  7: 'Forensics',
+  8: 'Misc',
+  9: 'Start',
+  10: 'PCAP',
+  11: 'Coding',
+  12: 'Mobile',
+  13: 'OSINT',
+  14: 'Blockchain',
+  15: 'Hardware',
+  16: 'Warmup',
+  17: 'Attack',
+  18: 'Defence',
+  20: 'Cloud',
+  21: 'Scada',
+};
+
+const HTBParser: Parser = {
+  name: 'HTB parser',
+  hint: 'paste https://ctf.hackthebox.com/api/ctf/<ctfid> from the network tab',
+
+  parse(s: string): ParsedTask[] {
+    const tasks = [];
+    const data = parseJsonStrict<{
+      challenges: Array<{
+        id: number;
+        name: string;
+        description: string;
+        challenge_category_id: number;
+      }>;
+    }>(s);
+    if (data == null) {
+      return [];
+    }
+
+    for (const challenge of data.challenges) {
+      if (
+        !challenge.description ||
+        !challenge.name ||
+        !challenge.challenge_category_id
+      ) {
+        continue;
+      }
+
+      let category = challengeCategories[challenge.challenge_category_id];
+      if (category == null) category = 'Unknown';
+
+      tasks.push({
+        title: challenge.name,
+        category: category,
+        description: challenge.description,
+      });
+    }
+    return tasks;
+  },
+  isValid(s) {
+    const data = parseJson<{
+      challenges: Array<{
+        id: number;
+        name: string;
+        description: string;
+        challenge_category_id: number;
+      }>;
+    }>(s);
+    return data != null;
```

Comment:
I'm confused. By reading this code, I assume it's supposed to return false if the JSON does not match a specific schema.

When I try to import a ctfd JSON, it sets data to (instead of `null`) :
```json
{
	"success": true,
	"data": [{
		"category":"recon",
		"name":"Welcome",
		"template":"/plugins/dynamic_challenges/assets/view.html",
		"tags":[],
		"script":"/plugins/dynamic_challenges/assets/view.js",
		"type":"dynamic",
		"id":17,
		"value":100
	}]
}
```

As a result, *any* valid JSON is considered as an HTB JSON, including `[]`, `42`, `false`.

Reply:
I was thinking that the `parseJson` implementation tries to parse the JSON as the type I gave it and if that fails, it would return `null`. But that assumption is completely wrong! :laughing: 

The issue should now be properly fixed.